### DISCUSSION TITLE
Scenario dependable physics

### DISCUSSION THREAD

**Discussion by karst** [OpenRCT2/OpenRCT2#18712] [Feature requests]
I think it'd be a very welcome addition to have new scenario options available that completely change the physics of the game.
Changing the physics of rides will make it difficult to import existing tracks, as it's possible the designs will break with different physics. I do think with NTSF it should save what physics is used and until then it should display a warning message before loading a coaster with different physics `Warning: The coaster you are loading is using <Physics type>, it's possible the coaster will not complete its circuit` with `<Physics type>` in this case being RCT2 vanilla. 
With a drop down menu we could have a few gravity physics presets: `RCT2`, `Realistic`, `Moon`, `Toy House / Minature`, `Zero-G`, or something along those lines. It would another line of depth to the game and could make for very interesting challenges. You shouldn't be able to save TD6 with different a physics type than RCT2's. 

This would also allow some physics related bugs to be patched, like vehicle spacing on the vertical axis, without invalidating over 20 years of coaster designs.

**Comment by ocalhoun6** [comment]
I like it, especially for scenarios that are supposed to happen on different planets.

Possibly ride friction could also be changed, thus simulating different amounts of air resistance.

I'd also love to see it be able to go all the way to *zero* gravity, which will then present interesting challenges for some ride designs. Gravity based coasters that get their energy from a lift hill would be almost useless, as they'd never go faster than the chain lift speed. Only coasters with launched starts and/or booster pieces would be of any real use. It would also make certain rides (like the launched freefall) more difficult or even utterly impossible to build. And, of course, you'd have to be *extremely* careful with any ride type that could fly off the track.

It would also be nice if the gravity setting affected the peep fall speed. (Though you'd have to cheat with that somewhat to make sure they still fall *a little* even in zero gravity, since it would be game-breaking if they didn't move at all while falling.) That's of no *practical* use, but it would just be a neat little feature, and it would help give the player a sense for what the gravity is like.

-----

While we're at it, it would probably be good to have a 'no weather' climate type for use in extraterrestrial parks. It's a bit weird when your park is supposedly on the moon, and then it starts to rain, as it always *eventually* will, even in the 'hot and dry' climate type.

The 'no weather' climate type would also be useful for sandbox builders making showcase parks, since they'd no longer have to worry about how their park will look during rain.

It could also be of some use when using income from shop sales as a victory condition, since it means players won't be able to use the umbrella exploit for massive amounts of shop sales when it rains.

----

I don't know if all the TD6 restrictions are necessary, though. The physics type shouldn't need to be saved with the ride, except perhaps to show a warning to players when they're about to build one that doesn't match the physics in their current scenario. (Much like the warning it displays if you don't have the same vehicle type as the ride was saved with.)

Other than that, let the player do what they will. They should know that if they try to build a normal-gravity coaster in a high- or low-gravity scenario, then that coaster likely won't work properly. Likewise, if they build a design in an unusual gravity environment, they should know that this new design probably won't work well in ordinary Earth gravity. *But* let them do it anyway if they choose. Heck, a big part of the fun of having different levels of gravity might be found in taking normal-gravity track designs and seeing how they behave with different amounts of gravity.

There *would* need to be some kind of on-screen indicator for this, though, so that the player *knows* what kind of gravity they're dealing with. Maybe something alongside/next to the weather report?

**Comment by James103** [comment]
Could ride stat calculations also be changed based on the current physics setting? If so, I'm imagining something like the following:
* `RCT2`, `Classic`, or `Original`: Original physics and ride stat calculation algorithms from RCT2 are used.
* `Realistic`: More realistic physics and ride stat calculations.
* `Hard`: Same as `Realistic`, but some ride stat calculations are changed to make it harder.
* `Moon`, `Mars`, `Jupiter`, etc: Same as `Realistic`, but physics are adjusted with respect to the gravity at the surface of the named celestial body.
